src/pretix/control/forms/orders.py

```diff
@@ -118,3 +118,9 @@ class OrderContactForm(forms.ModelForm):
     class Meta:
         model = Order
         fields = ['email']
+
+
+class OrderLocaleForm(forms.ModelForm):
+    class Meta:
+        model = Order
+        fields = ['locale']
```

Comment:
Given an event ``event``, ``event.settings.locales`` is a list of languages that are supported for this event.

You could do for example the following (untested):

    class OrderLocaleForm(forms.ModelForm):
        class Meta:
            model = Order
            fields = ['locale']

        def __init__(self, *args, **kwargs):
            super().__init__(*args, **kwargs)
            self.fields['locale'].choices = self.instance.event.settings.locales